test/cpp/ext/otel/otel_plugin_test.cc

```diff
@@ -757,6 +757,142 @@ TEST_F(OpenTelemetryPluginEnd2EndTest,
+                                       grpc::OpenTelemetryPluginBuilder::
+                                           kClientAttemptDurationInstrumentName,
+                                       grpc::OpenTelemetryPluginBuilder::
+                                           kServerCallStartedInstrumentName,
+                                       grpc::OpenTelemetryPluginBuilder::
+                                           kServerCallDurationInstrumentName})
+                    .add_optional_label("grpc.lb.locality")
+                    .set_labels_to_inject(
+                        {{grpc_core::ClientCallTracer::CallAttemptTracer::
+                              OptionalLabelKey::kLocality,
+                          grpc_core::RefCountedStringValue("locality")}})));
+  SendRPC();
+  auto data = ReadCurrentMetricsData(
+      [&](const absl::flat_hash_map<
+          std::string,
+          std::vector<opentelemetry::sdk::metrics::PointDataAttributes>>&
+              data) {
+        return !data.contains("grpc.client.attempt.started") ||
+               !data.contains("grpc.client.attempt.duration") ||
+               !data.contains("grpc.server.call.started") ||
+               !data.contains("grpc.server.call.duration");
+      });
+  // Verify client side metric (grpc.client.attempt.started) does not sees this
+  // label
+  ASSERT_EQ(data["grpc.client.attempt.started"].size(), 1);
+  const auto& client_attributes =
+      data["grpc.client.attempt.started"][0].attributes.GetAttributes();
+  EXPECT_THAT(
+      client_attributes,
+      ::testing::Not(::testing::Contains(::testing::Key("grpc.lb.locality"))));
+  // Verify client side metric (grpc.client.attempt.duration) sees this label.
+  ASSERT_EQ(data["grpc.client.attempt.duration"].size(), 1);
+  const auto& client_duration_attributes =
+      data["grpc.client.attempt.duration"][0].attributes.GetAttributes();
+  EXPECT_EQ(
+      absl::get<std::string>(client_duration_attributes.at("grpc.lb.locality")),
+      "locality");
+  // Verify server metric (grpc.server.call.started) does not see this label
+  ASSERT_EQ(data["grpc.server.call.started"].size(), 1);
+  const auto& server_attributes =
+      data["grpc.server.call.started"][0].attributes.GetAttributes();
+  EXPECT_THAT(
+      server_attributes,
+      ::testing::Not(::testing::Contains(::testing::Key("grpc.lb.locality"))));
+  // Verify server metric (grpc.server.call.duration) does not see this label
+  ASSERT_EQ(data["grpc.server.call.duration"].size(), 1);
+  const auto& server_duration_attributes =
+      data["grpc.server.call.duration"][0].attributes.GetAttributes();
+  EXPECT_THAT(
+      server_duration_attributes,
+      ::testing::Not(::testing::Contains(::testing::Key("grpc.lb.locality"))));
+}
+
+// Tests that when locality label is enabled on the plugin but not provided by
+// gRPC, an empty value is recorded.
+TEST_F(OpenTelemetryPluginEnd2EndTest,
+       OptionalPerCallLocalityLabelWhenNotAvailable) {
```

Comment:
Please also add a test for the case where the label is added to the call tracer but not enabled via the OT plugin.

Reply:
added